services/schema.service.ts

```diff
@@ -2,23 +2,29 @@ import { Injectable } from '@angular/core';
 import { Dataset } from '../interfaces/api.interfaces';
 import { FormBuilder, FormGroup, Validators, ValidatorFn } from '@angular/forms';
 
-import { Package, DataPackage, Field } from 'datapackage';
+import { Package, DataPackage, Schema, Field } from 'datapackage';
 
 import { Observable } from 'rxjs/Observable';
 import { fromPromise } from 'rxjs/observable/fromPromise';
 import { zip } from 'rxjs/observable/zip';
 import { map } from 'rxjs/operators';
 
-import { FormDescriptor } from '../interfaces/form.interfaces';
+import { FieldDescriptor, FieldOption, FormDescriptor, ObservationFormDescriptor } from '../interfaces/form.interfaces';
 
 @Injectable()
 export class SchemaService {
     private static readonly NON_VALIDATED_SCHEMA_CONSTRAINTS = ['unique', 'enum'];
 
+    private static readonly LOCATION_FIELDS = ['datum', 'lat', 'long', 'lon', 'latitude', 'longitude', 'accuracy'];
+
     private static schemaFieldTypeToFormFieldType(field: Field): string {
         if (field.type === 'string') {
             if ('constraints' in field && 'enum' in field.constraints) {
-                return 'select';
+                if (field.constraints.enum.length === 1) {
```

Comment:
I would argue that the field should also be set as required to be a automatic/hidden field.
We will discuss about that after.